platforms/core-configuration/declarative-dsl-core/src/main/kotlin/org/gradle/internal/declarativedsl/dom/resolution/DocumentResolutionContainer.kt

```diff
@@ -29,6 +29,6 @@ import org.gradle.internal.declarativedsl.dom.data.ValueDataContainer
 
 interface DocumentResolutionContainer :
     NodeDataContainer<DocumentNodeResolution, ElementResolution, PropertyResolution, ErrorResolution>,
-    ValueDataContainer<ValueNodeResolution, ValueFactoryResolution, LiteralValueResolved>
+    ValueDataContainer<ValueNodeResolution, ValueFactoryResolution, LiteralValueResolved, ValueNodeResolution.NamedReferenceResolved>
```

Comment:
The last type argument should be an interface that represents either a successful resolution or a resolution error, just as with value factories.